src/components/Images/Images.js

```diff
@@ -40,125 +52,105 @@ export const CloudinaryImage = ({
+
+  const buildImageTag = () => {
+    const format = fileFormats.slice(-1)[0]
+    let imageSettings = {
+      ...baseImageSettings,
+      format: format,
+      dpr: '1.0',
+    }
 
-  let fileFormats = ['webp', 'jp2', 'jpeg']
-
-  let dprSettings = ['1.0', '2.0', '3.0']
-
-  let mediaBreakpoints = [
-    Media.BREAKPOINTS.DESKTOP_RANGE_START,
-    Media.BREAKPOINTS.LAPTOP_RANGE_START,
-    Media.BREAKPOINTS.TABLET_RANGE_START,
-    Media.BREAKPOINTS.PHONE_RANGE_END,
-  ]
-
-  // We are expecting width/height attribute arrays in the order of
-  // Phone/Tablet/Laptop/Desktop but we have to setup media queries
-  // in the opposite order, so we reverse the arrays here.
-  if (width) {
-    width.reverse()
-  }
-  if (height) {
-    height.reverse()
+    return(
+      <img
+        key={`${uuidv4()}`}
+        data-src={cld.url(filePath(),imageSettings)}
+        className={imageClasses.join(' ')}
+        alt={alt}
+      />
+    )
   }
 
-  let sourceTags = []
-  let srcSetAttribute = lazyload ? 'data-srcset' : 'srcSet'
-  let srcAttribute = lazyload ? 'data-src' : 'src'
+  const buildSrcTags = (format) => {
+    const dprSettings = ['1.0', '2.0', '3.0']
+    let sourceTags = []
 
-  for (let format = 0; format < fileFormats.length; format++) {
+    // We are expecting width/height attribute arrays in the order of
+    // Phone/Tablet/Laptop/Desktop but we have to setup media queries
+    // in the opposite order, so we reverse the arrays here.
+    if (width) {
+      width.sort().reverse()
+    }
+    if (height) {
+      height.sort().reverse()
+    }
+    
     for (
       let breakpoint = 0;
       breakpoint < mediaBreakpoints.length;
       breakpoint++
-    ) {
-      let srcsetData = []
+      ) {
 
-      for (let dpr = 0; dpr < dprSettings.length; dpr++) {
-        let imageSettings = {
+      const srcsetData = dprSettings.map((dpr, indx)=>{
+        const imageSettings = {
           ...baseImageSettings,
-          ...(!!width[breakpoint] && { width: width[breakpoint] }),
-          ...(!!height[breakpoint] && { height: height[breakpoint] }),
-          format: fileFormats[format],
-          dpr: dprSettings[dpr],
+          ...(width && !!width[breakpoint] && { width: width[breakpoint] }),
+          ...(height && !!height[breakpoint] && { height: height[breakpoint] }),
+          format: format,
+          dpr: dpr,
         }
 
-        srcsetData.push(
-          cld.url(publicIdFilename, imageSettings) + ` ${dpr + 1}x`
-        )
-      }
-      if (
-        format === fileFormats.length - 1 &&
-        breakpoint === mediaBreakpoints.length - 1
-      ) {
-        let imageSettings = {
-          ...baseImageSettings,
-          format: fileFormats[format],
-          dpr: '1.0',
-        }
+        return cld.url(filePath(), imageSettings) + ` ${indx + 1}x`
+      })
+
+      let minMax = breakpoint < mediaBreakpoints.length - 1 ? `min` : `max`
+      sourceTags.push(<source
```

Comment:
You can use `const` if you're only pushing and not literally overwriting this

Reply:
constantized